sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+              "ElasticSearchTcpImpl:update: update was not success:" + response.getResult(),
+              LoggerEnum.INFO.name());
+        }
+      } catch (Exception e) {
+        ProjectLogger.log(
+            "ElasticSearchTcpImpl:update: exception occured:" + e.getMessage(),
+            LoggerEnum.ERROR.name());
+        promise.failure(e);
+      }
+    } else {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:update: Requested data is invalid.", LoggerEnum.INFO.name());
+    }
+    long elapsedTime = ElasticSearchHelper.calculateEndTime(startTime);
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:update: method end  =="
+            + " for Index "
+            + index
+            + " ,Total time elapsed = "
+            + elapsedTime,
+        LoggerEnum.PERF_LOG);
+    promise.success(false);
+    return promise.future();
+  }
+
+  /**
+   * This method will upsert data based on identifier.take the data based on identifier and merge
+   * with incoming data then update it and if identifier does not exist , it will insert data .
+   *
+   * @param index String
+   * @param identifier String
+   * @param data Map<String,Object>
+   * @return boolean
+   */
+  @Override
+  public Future<Boolean> upsert(String index, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    Promise<Boolean> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:upsert: method started at ==" + startTime + " for INdex " + index,
+        LoggerEnum.PERF_LOG);
+    if (!StringUtils.isBlank(index)
+        && !StringUtils.isBlank(identifier)
+        && data != null
+        && data.size() > 0) {
+      IndexRequest indexRequest = new IndexRequest(index, _DOC, identifier).source(data);
+      UpdateRequest updateRequest =
+          new UpdateRequest(index, _DOC, identifier).doc(data).upsert(indexRequest);
+      UpdateResponse response = null;
+      try {
+        response = ConnectionManager.getClient().update(updateRequest).get();
+      } catch (InterruptedException | ExecutionException e) {
+        ProjectLogger.log(e.getMessage(), e);
+        promise.failure(e);
+        promise.success(false);
+        return promise.future();
+      }
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:upsert: updated response==" + response.getResult().name());
+      if (ElasticSearchHelper.upsertResults.contains(response.getResult().name())) {
+        long elapsedTime = ElasticSearchHelper.calculateEndTime(startTime);
+        ProjectLogger.log(
+            "ElasticSearchTcpImpl:upsert: method end =="
+                + " for index "
+                + index
+                + " ,Total time elapsed = "
+                + elapsedTime,
+            LoggerEnum.PERF_LOG);
+        promise.success(true);
+        return promise.future();
+      }
+    } else {
+      ProjectLogger.log("ElasticSearchTcpImpl:upsert: Requested data is invalid.", LoggerEnum.INFO);
```

Comment:
change LoggerEnum.INFO to LoggerEnum.INFO.name()

Reply:
Done